responsemanager/selectorvalidator/selectorvalidator.go

```diff
@@ -0,0 +1,78 @@
+package selectorvalidator
+
+import (
+	"errors"
+
+	"github.com/ipfs/go-graphsync/ipldbridge"
+	ipld "github.com/ipld/go-ipld-prime"
+	ipldfree "github.com/ipld/go-ipld-prime/impl/free"
+	"github.com/ipld/go-ipld-prime/traversal"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
+)
+
+var (
+	// ErrInvalidLimit means this type of recursive selector limit is not supported by default
+	// -- to prevent DDOS attacks
+	ErrInvalidLimit = errors.New("unsupported recursive selector limit")
+)
+
+// ValidateSelector applies the default selector validation policy to a selector
+// on an incoming request -- which by default is to limit recursive selectors
+// to a fixed depth
+func ValidateSelector(bridge ipldbridge.IPLDBridge, node ipld.Node, maxAcceptedDepth int) error {
+	ssb := builder.NewSelectorSpecBuilder(ipldfree.NodeBuilder())
+
+	// this selector is a selector for traversing selectors...
+	// it traverses the various selector types looking for recursion limit fields
+	// and matches them
+	s, err := ssb.ExploreRecursive(selector.RecursionLimitNone(), ssb.ExploreFields(func(efsb builder.ExploreFieldsSpecBuilder) {
```

Comment:
I think we'll have to revisit this later to take fan-out into account. 100 is small for a chain, but possibly large for a large fanout.

However, we can handle this later. This validation is strictly better than what we have and is a clean way to throw out known-bad selectors.